Meadowlark-js/packages/meadowlark-core/src/validation/SchoolYearValidator.ts

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import { EnvironmentVariable, getFromCache, getValueFromEnvironment, updateCache } from '../Environment';
+import { SchoolYearEnumerationDocument } from '../model/SchoolYearEnumerationDocument';
+import { SchoolYearReferenceDocument } from '../model/SchoolYearReferenceDocument';
+
+const ALLOWED_SCHOOL_YEARS: EnvironmentVariable = 'ALLOWED_SCHOOL_YEARS';
+
+/**
+ * List of allowed school years.
+ * @returns
+ */
+export function getAllowedSchoolYears(): number[] {
+  const cached = getFromCache<[number]>(ALLOWED_SCHOOL_YEARS);
+  if (cached) {
+    return cached;
+  }
+
+  const config = getValueFromEnvironment(ALLOWED_SCHOOL_YEARS);
+
+  const list = config
+    .split(',')
+    .map((y: string) => (/^\d+$/.test(y.trim()) ? Number.parseInt(y, 10) : NaN))
+    .filter((n: number) => !Number.isNaN(n));
+
+  updateCache(ALLOWED_SCHOOL_YEARS, list);
+
+  return list;
```

Comment:
Might be worthwhile, at some point, to refactor this so that some of this code moves into the `Environment` module, perhaps with a signature like:

```
getConfigValue<T>(key, mappingFunction): T
```